### PR TITLE
Add dummy LRO service

### LINKED ISSUE #180: Consider adding a dummy LRO service
Can we add a simple service that implements [operations.proto](https://github.com/googleapis/googleapis/blob/master/google/longrunning/operations.proto) as an optional part of showcase (not in echo). It doesn't need to actually manage operations but we'd like it to be available so we can exercise the generated helper methods for accessing these methods.

See https://github.com/googleapis/gapic-generator-ruby/issues/132 for more context.

### PR DESCRIPTION
**Pull request by jbolinger**
Very simple implementation of the remaining LRO methods.

Note that the LRO methods don't appear to be a part of the CLI and I did not attempt to add them.

[resolves #180]

### CODE REVIEW

File: server/services/operations_service.go [hunk 1]
```diff
@@ -125,18 +128,70 @@ func searchFilterFunc(query string) func(b *pb.Blurb) bool {
 	}
 }
 
+// returns a successful response if the resource name is not blank
```
Comment (noahdietz) [inline server/services/operations_service.go]: A godoc comment must begin with the method name

`// CancelOperation ...`

File: server/services/operations_service.go [hunk 2]
```diff
@@ -125,18 +128,70 @@ func searchFilterFunc(query string) func(b *pb.Blurb) bool {
...
 func (s operationsServerImpl) CancelOperation(ctx context.Context, in *lropb.CancelOperationRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "google.longrunning.CancelOperation is unimplemented.")
+	if in.Name == "" {
+		return nil, status.Error(codes.NotFound, "cannot cancel operation without a name.")
+	}
+	return &empty.Empty{}, nil
 }
 
+// returns a fixed response matching the given PageSize if the resource name is not blank
```
Comment (noahdietz) [inline server/services/operations_service.go]: Same thing regarding godoc comments needing to start with the method name.

File: server/services/operations_service.go [hunk 3]
```diff
@@ -125,18 +128,70 @@ func searchFilterFunc(query string) func(b *pb.Blurb) bool {
...
+		})
+	}
+
+	return &lropb.ListOperationsResponse{
+		Operations:    operations,
+		NextPageToken: "txenNext",
+	}, nil
 }
 
+// returns a successful response if the resource name is not blank
```
Comment (noahdietz) [inline server/services/operations_service.go]: Same thing regarding godoc comments needing to start with the method name.

File: server/services/operations_service.go [hunk 4]
```diff
@@ -125,18 +128,70 @@ func searchFilterFunc(query string) func(b *pb.Blurb) bool {
...
 func (s operationsServerImpl) DeleteOperation(ctx context.Context, in *lropb.DeleteOperationRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "google.longrunning.DeleteOperation is unimplemented.")
+	if in.Name == "" {
+		return nil, status.Error(codes.NotFound, "cannot delete operation without a name.")
+	}
+	return &empty.Empty{}, nil
 }
 
+// randomly waits and returns an operation with the same name
```
Comment (noahdietz) [inline server/services/operations_service.go]: Same thing regarding godoc comments needing to start with the method name.

File: server/services/operations_service.go [hunk 5]
```diff
@@ -125,18 +128,70 @@ func searchFilterFunc(query string) func(b *pb.Blurb) bool {
...
 func (s operationsServerImpl) WaitOperation(ctx context.Context, in *lropb.WaitOperationRequest) (*lropb.Operation, error) {
-	return nil, status.Error(codes.Unimplemented, "google.longrunning.WaitOperation is unimplemented.")
+	if in.Name == "" {
+		return nil, status.Error(codes.NotFound, "cannot wait on a operation without a name.")
+	}
+
+	num := rand.Intn(500)
+	time.Sleep(time.Duration(num) * time.Millisecond)
```
Comment (noahdietz) [inline server/services/operations_service.go]: Why not use the timeout in [lropb.WaitRequest](https://github.com/googleapis/go-genproto/blob/fa694d86fc64c7654a660f8908de4e879866748d/googleapis/longrunning/operations.pb.go#L400)? The page size was used in the List RPC. I have a feeling folks would expect the timeout to be respected. WDYT?

File: server/services/operations_service.go (hunk 1 above)
Comment (jbolinger) [reply server/services/operations_service.go]: TIL - thanks! done.

File: server/services/operations_service.go (hunk 3 above)
Comment (jbolinger) [reply server/services/operations_service.go]: done

File: server/services/operations_service.go (hunk 4 above)
Comment (jbolinger) [reply server/services/operations_service.go]: done

File: server/services/operations_service.go (hunk 2 above)
Comment (jbolinger) [reply server/services/operations_service.go]: done

File: server/services/operations_service.go (hunk 5 above)
Comment (jbolinger) [reply server/services/operations_service.go]: I don't really have a preference. The [`WaitOperation`](https://github.com/googleapis/googleapis/blob/2a2c5518f64010c4e458afc818e57ed24fecdf6d/google/longrunning/operations.proto#L116) is a little strange anyway. The docs basically say that it's free to ignore the timeout and it doesn't even have to wait. So, random seemed fitting to me.

Comment (noahdietz) [reply server/services/operations_service.go]: Ok fine by me

**Pull request merged**